src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,205 @@
+import {
+    cloneElement,
+    HTMLAttributes,
+    useEffect,
+    useRef,
+    useState,
+} from "react";
+import styled from "styled-components";
+import useOutsideClick from "../../hooks/useOutsideClick";
+import { SVGProps } from "../../icons";
+
+type PositionProps = "top" | "bottom" | "left" | "right";
+
+type ElementInfo = {
+    width: number;
+    height: number;
+};
+
+export type PopoverOptionProps = {
+    icon?: (props: SVGProps) => JSX.Element;
+    label: string;
+    callback: VoidFunction;
+};
+
+export type PopoverProps = HTMLAttributes<HTMLDivElement> & {
+    keepOpenOnClick?: boolean;
+    options: PopoverOptionProps[];
+    position?: PositionProps;
+    onHover?: boolean;
+};
+
+const PopoverBox = styled.div<{
+    show: boolean;
+    position: PositionProps | undefined;
+    childInfo: ElementInfo;
+    popoverInfo: ElementInfo;
+}>`
+    opacity: 1;
+    pointer-events: auto;
+    background-color: rgb(255 255 255/1);
+    border-radius: 0.75rem;
+    box-shadow: 0 12px 48px -12px rgb(0 0 0 / 24%);
+    flex-direction: column;
+    gap: 0.25rem;
+    min-width: 15rem;
+    padding: 0.5rem;
+    position: absolute;
+    transition: all 0.2s ease-in-out 0s;
+    transition-property: all;
+    transition-duration: 0.2s;
+    transition-timing-function: ease-in-out;
+    transition-delay: 0s;
+    ${(props) => {
+        if (props.show) {
+            return `
+                opacity: 1;
+                pointer-events: auto;
+            `;
+        }
+
+        return `
+                opacity: 0;
+                pointer-events: none;
+           `;
+    }}
+
+    ${(props) => {
+        if (props.position && props.position !== "bottom") {
+            if (props.position === "top") {
+                return `
+                    margin-top: -${props.popoverInfo.height + props.childInfo.height}px;
+                `;
+            }
+
+            if (props.position === "left") {
+                return `
+                    margin-left: -${props.popoverInfo.width - 2}px;
+                `;
+            }
+
+            if (props.position === "right") {
+                return `
+                    margin-left: ${props.childInfo.width - 2}px;
+                `;
+            }
+        }
+
+        return `
+            margin-top: ${props.childInfo.height}px;
+           `;
+    }}
+`;
```

Comment:
This would be cleaner with a `switch` case. 
See `Button` component for example.